kdc/kerberos5.c

```diff
@@ -2032,6 +2032,59 @@ add_enc_pa_rep(astgs_request_t r)
 			   KRB5_PADATA_FX_FAST, NULL, 0);
 }
 
+static krb5_error_code
+add_supported_enctypes(astgs_request_t r)
+{
+    krb5_error_code ret;
+    uint32_t *supported_enctypes = NULL;
+    uint32_t bit = 1;
+    Key *key = NULL;
+
+    static const krb5_enctype enctypes[] = {
+	ETYPE_DES_CBC_CRC,
+	ETYPE_DES_CBC_MD5,
+	ETYPE_ARCFOUR_HMAC_MD5,
+	ETYPE_AES128_CTS_HMAC_SHA1_96,
+	ETYPE_AES256_CTS_HMAC_SHA1_96,
+	ETYPE_NULL
+    };
+    const krb5_enctype *p = enctypes;
+
+    if (!r->ek.encrypted_pa_data) {
```

Comment:
Given this is also done by `add_enc_pa_rep()`, perhaps we need a helper version of `krb5_padata_add()` that checks and adds this. e.g. `add_enc_pa_data()` or `add_encrypted_pa_data()`.